source/Xunit.Gherkin.Quick/ScenarioXunitHook/Scenario/ScenarioDiscoverer.cs

```diff
@@ -19,21 +19,26 @@ public ScenarioDiscoverer(IMessageSink messageSink)
             ITestMethod testMethod, 
             IAttributeInfo factAttribute)
         {
-            var feature = new FeatureDiscoveryModel(new FeatureFileRepository("*.feature")).Discover(testMethod.TestClass.Class.ToRuntimeType());
+            var featureClass = testMethod.TestClass.Class.ToRuntimeType();
+            var featureClassInfo = FeatureClassInfo.FromFeatureClassType(featureClass);
 
-            foreach (var scenario in feature.Children.OfType<global::Gherkin.Ast.Scenario>())
+            var features = new FeatureDiscoveryModel(new FeatureFileRepository(featureClassInfo.FileNameSearchPattern)).Discover(featureClass);
```

Comment:
This is what I mentioned earlier. This line creates tight coupling between this method and the FeatureDiscoveryModel. This method itself is not responsible to know about the FeatureClassInfo anything. Do that logic inside if necessary but not outside.